pkg/core/pipeline/source/main.go

```diff
@@ -236,14 +236,17 @@ func (s *Source) Unmarshal() (sourcer Sourcer, changelog Changelog, err error) {
 		changelog = &j
 
 	case "maven":
-		m := maven.Maven{}
-		err := mapstructure.Decode(s.Config.Spec, &m)
+		var sourceSpec maven.Spec
 
-		if err != nil {
+		if err := mapstructure.Decode(s.Config.Spec, &sourceSpec); err != nil {
 			return nil, nil, err
 		}
 
-		sourcer = &m
+		mavenResource, err := maven.New(sourceSpec)
+		if err != nil {
+			return nil, nil, err
+		}
+		sourcer = mavenResource
```

Comment:
Is `mavenResource` used elsewhere?
If not,
```suggestion
		sourcer, err := maven.New(sourceSpec)
		if err != nil {
			return nil, nil, err
		}
```

Reply:
for info @lemeurherve 

```
sourcer, err = maven.New(sourceSpec)
```

=> without the `:=` !

Rationale is that, if you use the `:=` operator, then it means that you re-declare the variable sourcer in this scope, which hides the initial declaration.
The value assignation will be lost (and both linter and compiler complains that the variable sourcer is not used anywhere, thanks to them :) )